File: lib/hello_world/hello_world_workflow.ex
defmodule HelloWorld.Workflow do
  use TemporalSdk.Workflow
  @moduledoc false

  @impl true
  def execute(_context, input) do
    a1 = start_activity(HelloWorld.Activity, [["hello"]])
    a2 = start_activity(HelloWorld.Activity, [["world"]])
    [%{result: a1_result}, %{result: a2_result}] = wait_all([a1, a2])
    IO.puts("#{a1_result} #{a2_result} #{input} \n")
  end
end

File: src/hello_world/hello_world_workflow.erl
-module(hello_world_workflow).

% elp:ignore W0012 W0040
-moduledoc false.

-export([
    execute/2
]).

-include_lib("temporal_sdk/include/workflow.hrl").

execute(_Context, Input) ->
    A1 = start_activity(hello_world_activity, [[~b"hello"]]),
    A2 = start_activity(hello_world_activity, [[~b"world"]]),
    [#{result := A1Result}, #{result := A2Result}] = wait_all([A1, A2]),
    io:fwrite("~s ~s ~s~n~n", [A1Result, A2Result, Input]).
